kong/plugins/session/storage/kong.lua

```diff
@@ -38,15 +38,36 @@ local function load_session_from_cache(key)
   return kong.cache:get(cache_key, nil, load_session_from_db, key)
 end
 
+local function insert_session_metadata(metadata, session)
+  if not metadata then
+    return
+  end
+
+  local audiences = metadata.audiences
+  local subjects  = metadata.subjects
+  local count     = #audiences
+  for i = 1, count do
+    kong.db.session_metadatas:insert({
```

Comment:
What if the insertion of session_metadatas get failed while the session insertion succeed?
Should we undo the insertion of session when it happens?

Reply:
I think so. Let me add an additional logic to cover it